int32 val

# success
int32 SUCCESS=1
int32 FAILURE=0
int32 NONE = -1

# general errors
int32 UNKNOWN_COMMAND=100
int32 TIMEOUT=101
int32 STATE_ERROR=102
int32 SHUTDOWN=103
int32 REQUIRED_DATA_MISSING=104

# motion failures
int32 PLANNING_FAILED=200
int32 CONTROL_FAILED=201
int32 NO_IK_SOLUTION=202
int32 INVALID_FRAME=203
int32 NO_GOAL_POSE=204


# transform failures
int32 TRANSFORM_POSE_FAILED=300

# dynamixel error
int32 DYNAMIXEL_ERROR = 400
int32 DYNAMIXEL_SEVERE_ERROR = 401

# object detection error
int32 NO_OBJECT_DETECTED = 500

# checks
int32 DEPTH_TOO_LOW = 600

# calibration error
int32 TAKE_SAMPLE_ERROR = 700 
